src/hooks/useENSSearch.ts

```diff
@@ -43,8 +56,23 @@ export default function useENSSearch({
       };
     }
 
-    const isAvailable = await getAvailable(name);
-    const rentPrice = await getRentPrice(name, duration);
+    const errors: (Error | undefined)[] = [];
+
+    const [
+      isAvailable,
+      rentPrice,
+      registrationDate,
+      nameExpires,
+    ] = await Promise.all([
+      getAvailable(name, contract).catch(e => (errors[0] = e)),
+      getRentPrice(name, duration, contract).catch(e => (errors[1] = e)),
+      fetchRegistrationDate(name + ENS_DOMAIN).catch(e => (errors[2] = e)),
+      getNameExpires(name).catch(e => (errors[3] = e)),
+    ]);
+
+    if (errors[0]) throw errors[0];
```

Comment:
can we not just do it the original way, but place the `getAvailable` and `getRentPrice` in a `Promise.all`?